egui/src/widgets/plot/items.rs

```diff
@@ -73,13 +73,317 @@ struct ExplicitGenerator {
+        Self {
+            shape: MarkerShape::Circle,
+            color: Color32::TRANSPARENT,
+            filled: true,
+            radius: 2.0,
+        }
+    }
+}
+
+impl Marker {
+    /// Get a vector containing a marker of each shape.
+    pub fn all() -> Vec<Self> {
+        vec![
+            Self::circle(),
+            Self::diamond(),
+            Self::square(),
+            Self::cross(),
+            Self::plus(),
+            Self::up(),
+            Self::down(),
+            Self::left(),
+            Self::right(),
+            Self::asterisk(),
+        ]
+    }
+
+    pub fn circle() -> Self {
+        Self {
+            shape: MarkerShape::Circle,
+            ..Default::default()
+        }
+    }
+
+    pub fn diamond() -> Self {
+        Self {
+            shape: MarkerShape::Diamond,
+            ..Default::default()
+        }
+    }
+
+    pub fn square() -> Self {
+        Self {
+            shape: MarkerShape::Square,
+            ..Default::default()
+        }
+    }
+
+    pub fn cross() -> Self {
+        Self {
+            shape: MarkerShape::Cross,
+            ..Default::default()
+        }
+    }
+
+    pub fn plus() -> Self {
+        Self {
+            shape: MarkerShape::Plus,
+            ..Default::default()
+        }
+    }
+
+    pub fn up() -> Self {
+        Self {
+            shape: MarkerShape::Up,
+            ..Default::default()
+        }
+    }
+
+    pub fn down() -> Self {
+        Self {
+            shape: MarkerShape::Down,
+            ..Default::default()
+        }
+    }
+
+    pub fn left() -> Self {
+        Self {
+            shape: MarkerShape::Left,
+            ..Default::default()
+        }
+    }
+
+    pub fn right() -> Self {
+        Self {
+            shape: MarkerShape::Right,
+            ..Default::default()
+        }
+    }
+
+    pub fn asterisk() -> Self {
+        Self {
+            shape: MarkerShape::Asterisk,
+            ..Default::default()
+        }
+    }
+
+    /// Set the marker's color. Defaults to the curve's color.
+    pub fn color(mut self, color: Color32) -> Self {
+        self.color = color;
+        self
+    }
+
+    /// Whether to fill the marker.
+    pub fn filled(mut self, filled: bool) -> Self {
+        self.filled = filled;
+        self
+    }
+
+    /// Set the maximum extent of the marker around its position.
+    pub fn radius(mut self, radius: f32) -> Self {
+        self.radius = radius;
+        self
+    }
+
+    pub(crate) fn get_shapes(&self, position: &Pos2) -> Vec<Shape> {
+        let sqrt_3 = 3f32.sqrt();
+        let frac_sqrt_3_2 = 3f32.sqrt() / 2.0;
+        let frac_1_sqrt_2 = 1.0 / 2f32.sqrt();
+
+        let Self {
+            color,
+            filled,
+            shape,
+            radius,
+        } = *self;
+
+        if color == Color32::TRANSPARENT {
+            return Vec::new();
+        }
+
+        let stroke_size = radius / 5.0;
+
+        let tf = |offset: Vec<Vec2>| -> Vec<Pos2> {
```

Comment:
this will cause a lot of extra allocations. Just make it a `|Vec2| -> Pos2` and call it per point instead